### PR TITLE
Use `curies.Converter` in Metadata object

### PR DESCRIPTION
**Pull request by cthoyt**
This is the next step in the process to switching everything to use converters.

### CODE REVIEW

File: src/sssom/typehints.py [hunk 1]
```diff
@@ -19,5 +24,33 @@
...
+
+        return cls(
+            converter=get_converter(),
+            metadata=get_default_metadata(),
+        )
+
+
+def generate_mapping_set_id() -> str:
+    """Getnerate a mapping set ID."""
+    return f"{SSSOM_URI_PREFIX}mappings/{uuid.uuid4()}"
```
Comment (matentzn) [inline src/sssom/typehints.py:48]: Note that this logic seems duplicated with https://github.com/mapping-commons/sssom-py/pull/421/files#diff-fa45af370531fd5131bc80371b323932f856b104add1550cb1e7fe281f821133R14

File: src/sssom/typehints.py [hunk 2]
```diff
@@ -19,5 +24,33 @@
...
+        from .context import get_converter
+
+        return cls(
+            converter=get_converter(),
+            metadata=get_default_metadata(),
+        )
+
+
+def generate_mapping_set_id() -> str:
+    """Getnerate a mapping set ID."""
```
Comment (matentzn) [inline src/sssom/typehints.py:47]:
```suggestion
    """Generate a mapping set ID."""
```

**Review by matentzn — COMMENTED** [review]
Two comments, tehn I will approve

File: src/sssom/typehints.py (hunk 1 above)
Comment (cthoyt) [reply src/sssom/typehints.py:48]: Resolved in 98122ac

**Review by matentzn — APPROVED** [review]

**Pull request merged**